src/MongoDB/Manager.c

```diff
@@ -669,11 +670,46 @@ static PHP_METHOD(Manager, startSession)
 		return;
 	}
 
-	if (options && php_array_exists(options, "causalConsistency")) {
+	if (options && php_array_existsc(options, "causalConsistency")) {
 		cs_opts = mongoc_session_opts_new();
 		mongoc_session_opts_set_causal_consistency(cs_opts, php_array_fetchc_bool(options, "causalConsistency"));
 	}
 
+	if (options && php_array_existsc(options, "defaultTransactionOptions")) {
+		zval* txn_options = php_array_fetch_array(options, "defaultTransactionOptions");
```

Comment:
Did you mean to use `php_array_fetchc_array` here? Though, see my comment below about restructuring before you address that.

Reply:
No longer relevant due to the previous comment.